contrib/endpoints/src/api_manager/config.cc

```diff
@@ -515,9 +515,17 @@ void Config::SetJwksUri(const string &issuer, const string &jwks_uri,
 
 std::string Config::GetFirebaseServer() {
   if (server_config_ == nullptr) {
+    if (service_.has_experimental() &&
+        service_.experimental().has_authorization() &&
+        service_.experimental().authorization().has_provider()) {
+      // In real environment, server_config_ should be null. And we rely on the
+      // field in service.
+      return service_.experimental().authorization().provider();
+    }
     return "";
   }
 
+  // It is the testing environment, use server config.
```

Comment:
By design, server_config over-ride service_config.  Please check server_config first.  

Server_config may have other fields,  only if it doesn't have firebase server field, check service_config.